Neos.Media/Classes/Domain/Model/AssetSource/Neos/NeosAssetProxy.php

```diff
@@ -0,0 +1,189 @@
+<?php
+namespace Neos\Media\Domain\Model\AssetSource\Neos;
+
+/*
+ * This file is part of the Neos.Media package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+  *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations\Inject;
+use Neos\Flow\Http\Uri;
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Neos\Flow\ResourceManagement\ResourceManager;
+use Neos\Media\Domain\Model\Asset;
+use Neos\Media\Domain\Model\AssetInterface;
+use Neos\Media\Domain\Model\AssetSource\AssetProxy\AssetProxy;
+use Neos\Media\Domain\Model\AssetSource\AssetSource;
+use Neos\Media\Domain\Model\ImageInterface;
+use Neos\Media\Exception\AssetServiceException;
+use Neos\Media\Exception\ThumbnailServiceException;
+use Psr\Http\Message\UriInterface;
+
+final class NeosAssetProxy implements AssetProxy
+{
+    /**
+     * @var NeosAssetSource
+     */
+    private $assetSource;
+
+    /**
+     * @var Asset
+     */
+    private $asset;
+
+    /**
+     * @Inject()
+     * @var PersistenceManagerInterface
+     */
+    protected $persistenceManager;
+
+    /**
+     * @Inject()
+     * @var ResourceManager
+     */
+    protected $resourceManager;
+
+    /**
+     * @param AssetInterface $asset
+     * @param NeosAssetSource $assetSource
+     */
+    public function __construct(AssetInterface $asset, NeosAssetSource $assetSource)
+    {
+        if (!$asset instanceof Asset) {
+            throw new \RuntimeException(sprintf('%s currently only support the concrete Asset implementation because several methods are not part of the AssetInterface yet.', __CLASS__), 1509635540408);
```

Comment:
Why don't we make it part of the interface?

Reply:
That would be a bigger breaking change. We should tackle this in a separate PR, to give it the required attention. I mean, sure, this would be a good time to do (for a major release). But time …